### DISCUSSION TITLE
Rendering of HTML elements in ngt-canvas

### DISCUSSION THREAD

**Discussion by SjorsJessen** [nartc/angular-three#9] [Ideas]
Hi,

First of all, thank you for creating this great library!
I was wondering if you were planning on making it possible to render HTML elements directly into the ngt-canvas (like how it's done in the react three drei library.) Not sure if this is already possible, but I couldn't find it in your documentation.

Would love to hear your thoughts.

Cheers!

**Comment by nartc** [comment]
I do plan to port Drei components over to https://github.com/nartc/angular-three/tree/main/packages/helpers but time has been a luxury for me. But to answer your question, yes, I do plan to support it as soon as I find some free time.

**Reply by nartc** [reply]
Of course, feel free to initiate with a PR. I'd love to receive contributions and I'm willing to guide/help with where to start and reviewing the code. If not, no worries :D

**Reply by SjorsJessen** [reply]
Thanks for the quick response! If I find the time myself I will try to port the HTML component. Not sure when, but I'll keep you posted if I have any updates.

**Reply by nartc** [reply]
@SjorsJessen I put together a hacky version of `Html` in `@angular-three/helpers/web` over the weekend. It is by no mean optimized and working correctly but check it out if you're interested.
![image](https://user-images.githubusercontent.com/25516557/124538926-19323400-dde2-11eb-84f4-b38bfb0e9f79.png)

![image](https://user-images.githubusercontent.com/25516557/124538885-028bdd00-dde2-11eb-83f5-d61e8a962ba7.png)

**Reply by nartc** [reply]
![html](https://user-images.githubusercontent.com/25516557/124630166-c5f2cc80-de47-11eb-94f8-7a8e7c15b95e.gif)

Cleaned up a little bit but the position of the Html seems off. If you could take a look at the code, It'd be appreciated

**Reply by SjorsJessen** [reply]
Thanks for taking the time to make this! Im going to take a look at it this evening.

**Reply by SjorsJessen** [reply]
Do you use these versions of your packages?
    "@angular-three/controls": "^1.0.0-beta.41",
    "@angular-three/core": "^1.0.0-beta.41",
    "@angular-three/helpers": "^1.0.0-beta.41",
    "@angular-three/postprocessing": "^1.0.0-beta.41",

**Reply by nartc** [reply]
> Do you use these versions of your packages?
> 
>     "@angular-three/controls": "^1.0.0-beta.41",
> 
>     "@angular-three/core": "^1.0.0-beta.41",
> 
>     "@angular-three/helpers": "^1.0.0-beta.41",
> 
>     "@angular-three/postprocessing": "^1.0.0-beta.41",

I just use the demo project in the main repo. Probably something missing? What's wrong?

**Reply by SjorsJessen** [reply]
Well, I updated @angular-three/helpers in my project to get access to the Html component. But after doing this I got a large amount of errors like these:

Error: ./node_modules/@angular-three/core/fesm2015/angular-three-core.js 1327:10-28
"export 'ɵɵFactoryTarget' (imported as 'i0') was not found in '@angular/core'

Error: ./node_modules/@angular-three/core/fesm2015/angular-three-core.js 1466:10-28
"export 'ɵɵFactoryTarget' (imported as 'i0') was not found in '@angular/core'

Error: ./node_modules/@angular-three/core/fesm2015/angular-three-core.js 1627:10-28
"export 'ɵɵFactoryTarget' (imported as 'i0') was not found in '@angular/core'

I'm trying to narrow down the problem, but it seems that when you update to ^1.0.0-beta.41 of your packages you'll get these issues.

**Reply by SjorsJessen** [reply]
Just used the demo you did with beeman.dev and installed the ^1.0.0-beta.41 in that project, which gave me the errors mentioned above. After downscaling back to @angular-three/core@1.0.0-beta.28 it solved the issues. 

Which version of Angular would you recommend to use with ^1.0.0-beta.41? Believe you are using version 12.0.4 right now (as mentioned in your repo)?

In the meantime I'll try some other things to get Html to work in a clean NG3 project.

**Reply by nartc** [reply]
well maybe I screwed up the release :D lemme try it on a clean proj as well

**Reply by nartc** [reply]
also make sure rxjs7 is installed

**Reply by nartc** [reply]
@SjorsJessen I just init a new project and it works fine.
![image](https://user-images.githubusercontent.com/25516557/124658721-ba180200-de69-11eb-8ac7-6c7c1f67ab94.png)

![image](https://user-images.githubusercontent.com/25516557/124658756-c734f100-de69-11eb-90db-600a408661d8.png)

![image](https://user-images.githubusercontent.com/25516557/124658781-cdc36880-de69-11eb-8ef5-b053bf47baf0.png)

**Reply by SjorsJessen** [reply]
![NG3](https://user-images.githubusercontent.com/12511561/124663790-d8eab880-deaa-11eb-936c-910e178cec58.PNG)

Thank you for your help, much appreciated!
